src/main/java/org/apache/datasketches/theta/HeapAlphaSketch.java

```diff
@@ -128,20 +128,19 @@ static HeapAlphaSketch heapifyInstance(final Memory srcMem, final long seed) {
     checkMemIntegrity(srcMem, seed, preambleLongs, lgNomLongs, lgArrLongs);
 
     final float p = extractP(srcMem);                             //bytes 12-15
-    final int lgRF = extractLgResizeFactor(srcMem);               //byte 0
-    final ResizeFactor myRF = ResizeFactor.getRF(lgRF);
+    final int memlgRF = extractLgResizeFactor(srcMem);            //byte 0
+    final ResizeFactor memRF = ResizeFactor.getRF(memlgRF);
 
     final double nomLongs = (1L << lgNomLongs);
     final double alpha = nomLongs / (nomLongs + 1.0);
     final long split1 = (long) (((p * (alpha + 1.0)) / 2.0) * LONG_MAX_VALUE_AS_DOUBLE);
 
-    if ((myRF == ResizeFactor.X1)
-            && (lgArrLongs != startingSubMultiple(lgNomLongs + 1, myRF, MIN_LG_ARR_LONGS))) {
-      throw new SketchesArgumentException("Possible corruption: ResizeFactor X1, but provided "
-              + "array too small for sketch size");
+    if (isResizeFactorIncorrect(srcMem, lgNomLongs, lgArrLongs)) {
+      throw new SketchesArgumentException("Possible corruption: ResizeFactor  "
+          + "inconsistent with lgNomLongs and lgArrLongs.");
```

Comment:
This is a different approach taken than that of the DirectQuickSelect sketch, where X2 is selected as a resize factor when the arguments are incorrect, as per the comment:
```//If incorrect it sets it to X2 which always works.```
Is this by design?

Reply:
Yes.  Look more closely at the object being modified in the case where X2 is set.  It is a WritableMemory.  In the case where  I have to throw, it is just Memory, which is immutable.  Big difference.